doc/user/compile.md

```diff
@@ -55,14 +55,14 @@ program (before including mlpack or Armadillo!).
 | `-DMLPACK_CERR_STREAM=std::cerr` | `#define MLPACK_CERR_STREAM std::cerr` | Set the default error stream.  (Defaults to `std::cerr`.) |
 | `-DMLPACK_PRINT_INFO` | `#define MLPACK_PRINT_INFO` | Print information messages (`[INFO ]`) during program execution. |
 | `-DMLPACK_PRINT_WARN` | `#define MLPACK_PRINT_WARN` | Print warning messages (`[WARN ]`) during program execution. |
-| `-DMLPACK_NO_STD_MUTEX` | `#define MLPACK_NO_STD_MUTEX` | Disable mutexes inside mlpack; use this if your system has no support for `std::mutex` and has only one core.  You may also need to define `ARMA_DO_NOT_USE_STD_MUTEX` for Armadillo. |
 | `-DMLPACK_SUPPRESS_FATAL` | `#define MLPACK_PRINT_FATAL` | Do not print `[FATAL]` messages during program execution. |
 | `-DENS_PRINT_INFO` | `#define ENS_PRINT_INFO` | Print informational messages from [ensmallen](https://www.ensmallen.org/) optimizers. |
 | `-DENS_PRINT_WARN` | `#define ENS_PRINT_WARN` | Print warning messages from [ensmallen](https://ensmallen.org/) optimizers. |
 |---------------------------|-------------------|---------------|
 |*Functionality.* |||
 | `-DMLPACK_ENABLE_ANN_SERIALIZATION` | `#define MLPACK_ENABLE_ANN_SERIALIZATION` | Allow neural network layers to be serialized. |
 | `-DMLPACK_DISABLE_STB` | `#define MLPACK_DISABLE_STB` | Disable [STB](https://github.com/nothings/stb)-related [image functionality](load_save.md#image-data). |
+| `-DMLPACK_NO_STD_MUTEX` | `#define MLPACK_NO_STD_MUTEX` | Disable mutexes inside mlpack; use this if your system has no support for `std::mutex` and has only one core.  You may also need to define `ARMA_DO_NOT_USE_STD_MUTEX` for Armadillo. |
```

Comment:
Just reading this and thinking ... should we not set it for armadillo if it is set for mlpack?

Reply:
The code does set it [when we can](https://github.com/mlpack/mlpack/pull/3868/files#diff-e9b6a3e1469948a43fad34b640cd9c71d8117c62b173f937ac9214687418e13fR81), but if the user included Armadillo before mlpack there isn't anything we can do.  Hence the "you may also need to" instead of "you must".  Maybe you have a clearer way to put it?

Reply:
@shrit change `ARMA_DO_NOT_USE_STD_MUTEX` to `ARMA_DONT_USE_STD_MUTEX` as per https://github.com/mlpack/mlpack/pull/3868/files#r2025876077